File: packages/Client/sdk/models/Member.ts
import { Entity, MemberTask, Post, MemberData, model } from 'sdk'

@model('Member')
export class Member extends Entity {
	role: 'Admin' | 'Member' = 'Member'
	nickname?: string
	password?: string
	token?: string
	imageId?: number
	tasks = new Array<MemberTask>()
	data = new Array<MemberData>()
	posts = new Array<Post>()
}

File: packages/Server/Application/Members/Member.cs

namespace Schulcast.Application.Members;

public struct MemberRoles
{
	public const string Admin = "Admin";
	public const string Member = "Member";
}

public class Member : Model
{
	[Required, MaxLength(50)] public string Nickname { get; set; } = null!;
	[MaxLength(50)] public string Role { get; set; } = MemberRoles.Member;
	[MaxLength(64)] public string? Password { get; set; }
	public int ImageId { get; set; }
	public List<MemberTask> Tasks { get; set; } = new();
	public List<MemberData> Data { get; set; } = new();
	public List<Post> Posts { get; set; } = new();
}